### ISSUE TITLE
Day one bugs

### ISSUE THREAD

**Issue by stanfordquan** [gslab-econ/template#100]
1) @arosenbe I found something that I think is potentially a day-one bug for the template's configuration.py. Line 47 and 48 will throw the following error if the config_user file is blank. I don't think this is the desired behavior. We can discuss follow ups, potentially creating a specialized error message for that case or simply warning the user of this possible error.

    ```
    PrerequisiteError: Inappropriate input values: `d` must be a dictionary, and at least one of `d[global][executable_names]` and `d[user][executable_names]` must be a dictionary as well.:
    ```

Add that we need to update config_global

2) please update the instruction in README.md on how to specify Stata executable name

3) Add warning that upgrading users need to update .gitignore

**Comment by arosenbe** [comment]
Hey @stanfordquan, do you have a blank config_global too? Here's the [traceback](https://github.com/gslab-econ/gslab_python/blob/bdce68d18e87cc2fb315d645e90bf81a2004885a/gslab_scons/misc.py#L242).

**Comment by stanfordquan** [comment]
Nope, config_global is good. I fixed it by editing config_user as said above.

**Comment by stanfordquan** [comment]
We successfully reproduced the bug. The bug is caused when config_global.yaml does not have this bit

```
# Executable names
executable_names:
  python: None
  r: None
  stata: None
  matlab: None
  lyx: none
  latex: None

# If True, check that application is installed/up-to-date before SCons run
prereq_checks:
  git_lfs: Yes
  gslab_python: Yes
  python: Yes
  r: No
  stata: No
  matlab: No
  lyx: No
  latex: No

```

Solution is to add a note to release note that users upgrading need to add that bit to their existing config_global.

**Comment by stanfordquan** [comment]
I'll edit the release notes and close this issue.

**Comment by stanfordquan** [comment]
Merged and closed in #101.
